### PR TITLE
feat: Swap service `update` signatures

### PR DESCRIPTION
**Pull request by tuliomir**
### Summary
This PR adds the option for the `/wallet/atomic-swap/tx-proposal/sign` route to also update the Atomic Swap Service with the signatures, besides generating the already expected `txHex`.

### Acceptance Criteria
- Adds the option to `update` the proposal signatures with the Atomic Swap Service

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/controllers/wallet/atomic-swap/tx-proposal.controller.js [hunk 1]
```diff
@@ -225,10 +225,39 @@ async function signTx(req, res) {
...
 
+    // Updating the service, if a proposal id was informed
+    if (constants.SWAP_SERVICE_FEATURE_TOGGLE && proposalId) {
+      // First, validate if the proposal is already registered with this wallet
+      const listenedProposals = await atomicSwapService.getListenedProposals(req.walletId);
+      if (!listenedProposals.has(proposalId)) {
+        res.status(404);
+        res.send({
+          success: false,
+          error: 'Proposal is not registered. Register it first through [POST] /register/:proposalId',
```
Comment (r4mmer) [inline src/controllers/wallet/atomic-swap/tx-proposal.controller.js:241]: The URL is wrong, we have some prefix (`/wallet/atomic-swap/...`).
Also, should we have this URL on the error message?

**Review by r4mmer — CHANGES_REQUESTED** [review]

File: src/services/atomic-swap.service.js [hunk 2]
```diff
@@ -74,6 +74,33 @@ const serviceCreate = async (walletId, partialTx, password) => {
...
+    throw new Error('Missing password');
+  }
+
+  const { success } = await swapService.update(updateParams);
+
+  if (!success) {
+    throw new Error('Unable to update the proposal on the Atomic Swap Service');
+  }
+
+  return { success };
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply src/services/atomic-swap.service.js]: Removed on [f655655](https://github.com/HathorNetwork/hathor-wallet-headless/pull/288/commits/f65565559bb66955c3a3bedde9c7ce6b443454b3).

File: src/services/atomic-swap.service.js [hunk 3]
```diff
@@ -74,6 +74,33 @@ const serviceCreate = async (walletId, partialTx, password) => {
...
+    throw new Error('Missing PartialTx');
+  }
+  if (!updateParams.password) {
+    throw new Error('Missing password');
+  }
+
+  const { success } = await swapService.update(updateParams);
+
+  if (!success) {
+    throw new Error('Unable to update the proposal on the Atomic Swap Service');
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply src/services/atomic-swap.service.js]: Changed to a more specific `SwapServiceError` on [f655655](https://github.com/HathorNetwork/hathor-wallet-headless/pull/288/commits/f65565559bb66955c3a3bedde9c7ce6b443454b3)

File: src/services/atomic-swap.service.js [hunk 4]
```diff
@@ -29,6 +29,17 @@ const {
  */
 const walletListenedProposals = new Map();
 
+/**
+ * Represents errors related to interactions with the Atomic Swap Service
+ */
+class SwapServiceError extends Error {
```
Comment (pedroferreira1) [inline src/services/atomic-swap.service.js:35]: I think would be better to have a separate file with the errors (and with a TODO to use the errors from the lib when you add it there)

And why don't you create a simple Error class like this?

```
class SwapServiceError extends Error {}
```

**Review by r4mmer — APPROVED** [review]

File: src/services/atomic-swap.service.js (hunk 4 above)
Comment (tuliomir) [reply src/services/atomic-swap.service.js]: This code was successfully rebased from 286

**Review by pedroferreira1 — APPROVED** [review]

**Pull request merged**